ccxt/async/exchange.py

```diff
@@ -72,6 +75,26 @@ def __del__(self):
     # def run_rest_poller_loop
     #     await asyncio.sleep (exchange.rateLimit / 1000.0)
 
+    async def wait_for_token(self):
+        while self.rateLimitTokens <= 1:
+            # if self.verbose:
+            #     print('Waiting for tokens: Exchange: {0}'.format(self.id))
+            self.add_new_tokens()
+            seconds_delays = [0.01, 0.1, 0.7, 1, 1.5, 2]
+            delay = random.choice(seconds_delays)
```

Comment:
@samholt, these two lines are not very clear to me, namely, the choice for 0.01, 0.1, 0.7, 1, 1.5 and 2. I can see that you are randomizing the delays here, but I'm not sure why. Can you please elaborate on that?

Reply:
Certainly, CS/EEE theory from how routers / ethernet optimally scales bandwidth to the maximal optimal rate for the resources. I.e. the Ethernet protocol or similarly the IEEE 802.3 CSMA/CD standard, both implement a exponential back off algorithm (https://en.wikipedia.org/wiki/Exponential_backoff) to avoid multiple accesses to the same resource (a collision) (This prevents hundreds of asyncio tasks checking for new available tokens at the same time). From theory the randomised delays should follow a similar process to https://en.wikipedia.org/wiki/Exponential_backoff#Example_exponential_backoff_algorithm, however a simplified version here is to just combine 1 - 3, to achieve an approximation, which is crudely the set above. Upon second thought the times should be exponentially distributed in the set, thus if we assume a minimum delay of 1ms and max delay of 1000ms, of a set of 5, thus: [0.001, 0.005, 0.022, 0.106, 0.5].

```
import numpy as np
x = np.linspace(np.log(0.001), np.log(0.5),5)
y = np.exp(x)

for i in y:
    print(round(i,3))
```